.github/workflows/push-packages.yml

```diff
@@ -0,0 +1,24 @@
+name: Publish to GitHub Packages
+on: push
+jobs:
+  publish:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Cache Maven artifacts
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
```

Comment:
This takes a hash coder from the pom.xml files and uses this has caching key to cache the .m2 folder. The cached files still have to be downloaded on every build so its probably not much faster, but at least this way the logs are not as messy :)